### PR TITLE
Больше подробностей (часть 1)

### PR DESCRIPTION
**Pull request by olegsorokin**
---
:mortar_board: [Больше подробностей (часть 1)](https://up.htmlacademy.ru/react/9/user/30818/tasks/6)

### CODE REVIEW

File: project/src/components/film-card/film-card.tsx [hunk 1]
```diff
@@ -1,19 +1,46 @@
 import { Link } from 'react-router-dom';
 import { AppRoute } from '../../constants/routs';
 import { Film } from '../../types/film';
+import { VideoPlayer } from '../video-player/video-player';
+import { useEffect, useState } from 'react';
```
Comment (dianatsevan) [inline project/src/components/film-card/film-card.tsx:5]: Let place imports in more consistent way: libs imports first, and our code further. Also, you might want to indent these imports with an empty line

File: project/src/components/film-card/film-card.tsx [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
   film: Film,
-  isActive: boolean,
-  onHover: (id: number) => void
 }
 
-function FilmCard({ film, isActive, onHover }: Props): JSX.Element {
-  const { id, poster, title } = film;
+function FilmCard({ film }: Props): JSX.Element {
+  const { poster, title, video } = film;
+  const [hovered, setHovered] = useState(false);
```
Comment (dianatsevan) [inline project/src/components/film-card/film-card.tsx:15]: Usually, I am using `is` prefix for my boolean vars. On the next line you have a var named with such a prefix. Don't you think, we need to follow common styles?

File: project/src/components/film-card/film-card.tsx [hunk 3]
```diff
@@ -1,19 +1,46 @@
...
+      clearTimeout(timeoutId);
+      setPlaying(false);
+    };
+  }, [hovered]);
 
   return (
-    <article className="small-film-card catalog__films-card" onMouseEnter={() => onHover(id)}>
+    <article
+      className="small-film-card catalog__films-card"
+      onMouseEnter={() => setHovered(true)}
```
Comment (dianatsevan) [inline project/src/components/film-card/film-card.tsx:36]: I prefer not to embed logic into layout, but embed references to the needed functions. What do you think about it?

**Review by dianatsevan — APPROVED** [review]

**Comment by dianatsevan** [comment]
Good job! Just a couple of minor comments. Let me know if you will change it and when I need to merge the PR

File: project/src/components/film-card/film-card.tsx (hunk 1 above)
Comment (olegsorokin) [reply project/src/components/film-card/film-card.tsx:5]: I added 1 empty line after libs imports and 2 empty lines after  list of all imports.

File: project/src/components/film-card/film-card.tsx (hunk 3 above)
Comment (olegsorokin) [reply project/src/components/film-card/film-card.tsx:36]: I like it!

File: project/src/components/film-card/film-card.tsx [hunk 4]
```diff
@@ -12,11 +14,19 @@ type Props = {
 
 function FilmCard({ film }: Props): JSX.Element {
   const { poster, title, video } = film;
-  const [hovered, setHovered] = useState(false);
+  const [isHovered, setHovered] = useState(false);
   const [isPlaying, setPlaying] = useState(false);
 
+  function handleMouseEnter() {
```
Comment (dianatsevan) [inline project/src/components/film-card/film-card.tsx:20]: Why it is not an arrow func?

File: project/src/components/film-card/film-card.tsx [hunk 5]
```diff
@@ -1,8 +1,10 @@
+import { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
+
 import { AppRoute } from '../../constants/routs';
 import { Film } from '../../types/film';
 import { VideoPlayer } from '../video-player/video-player';
-import { useEffect, useState } from 'react';
+
```
Comment (dianatsevan) [inline project/src/components/film-card/film-card.tsx:8]: 2 empty lines

File: project/src/components/film-list/film-list.tsx [hunk 6]
```diff
@@ -1,7 +1,9 @@
 import React from 'react';
+
 import { Film } from '../../types/film';
 import { FilmCard } from '../film-card/film-card';
 
+
```
Comment (dianatsevan) [inline project/src/components/film-list/film-list.tsx:6]: 2 empty lines

File: project/src/components/film-reviews/film-reviews.tsx [hunk 7]
```diff
@@ -7,6 +7,7 @@ import { Film } from '../../types/film';
 import { SimilarFilms } from '../similar-films/similar-films';
 import { IconAdd, IconPlayS } from '../icon';
 
+
```
Comment (dianatsevan) [inline project/src/components/film-reviews/film-reviews.tsx:10]: 2 empty lines

**Review by dianatsevan — APPROVED** [review]

**Pull request merged**